src/BookNavigator/BookNavigator.js

```diff
@@ -134,10 +135,22 @@ export class BookNavigator extends LitElement {
         bookContainerSelector: this.pageContainerSelector,
         bookreader: this.bookreader,
       }),
-      share: new SharingProvider(this.book.metadata, this.baseHost, this.itemType),
-      bookmarks: new BookmarksProvider(this.bookmarksOptions, this.bookreader)
+      share: new SharingProvider(this.book.metadata, this.baseHost, this.itemType, this.bookreader.options.subPrefix),
+      bookmarks: new BookmarksProvider(this.bookmarksOptions, this.bookreader),
     };
 
+    // add shortcut for volumes if multipleBooksList exists
+    if (this.bookreader.options.enableMultipleBooks) {
+      this.menuProviders.volumes = new VolumesProvider((brInstance) => {
```

Comment:
How come the `optionChange` callback passes back the same instance of bookreader that was passed in?

Reply:
yah, this callback is weird as currently all the providers have this `if (brInstance) { ... }` block.
for other providers that handle custom events like search inside, the brInstance gets updated & passed to handler via callback, so booknav's bookreader has to get updated. the bridge is not ideal, we are definitely looking at ways to improve this block.

this is mainly for parity to prepare for when we do dynamically swap books in-theater. 

sometim

Reply:
For the other callbacks that update the `brInstance`, is it passing a "new" bookreader instance or is it the same one?